### ISSUE TITLE
Export structured data from graph via js as CSV file

### ISSUE THREAD

**Issue by e22c** [goat1000/SVGGraph#58]
Is there an easy way to export structured graph data as CSV file from within graph via java script - eg. right click on graph and select export data or clicking on particular part of graph.

bellow and js example:

   let csv = arrayToCsv([
             [1, '2', '"3"'],
             [true, null, undefined],
            ]); 
   downloadBlob(csv, 'export.csv', 'text/csv;charset=utf-8;')

/** Convert a 2D array into a CSV string
 */
function arrayToCsv(data){
  return data.map(row =>
    row
    .map(String)  // convert every value to String
    .map(v => v.replaceAll('"', '""'))  // escape double colons
    .map(v => `"${v}"`)  // quote it
    .join(';')  // semi-comma-separated
  ).join('\r\n');  // rows starting on new lines
}

/** Download contents as a file
 * Source: https://stackoverflow.com/questions/14964035/how-to-export-javascript-array-info-to-csv-on-client-side
 */
function downloadBlob(content, filename, contentType) {
  // Create a blob
  var blob = new Blob([content], { type: contentType });
  var url = URL.createObjectURL(blob);

  // Create a link to download it
  var pom = document.createElement('a');
  pom.href = url;
  pom.setAttribute('download', filename);
  pom.click();
}

**Comment by goat1000** [comment]
No, there is not a simple way to do this. The easiest way is to add a link to a separate PHP script that would generate the CSV from the same data you pass into SVGGraph. You can probably find a decent CSV output library on Packagist to help with that.
